x/tokenfactory/types/keys.go

```diff
@@ -0,0 +1,44 @@
+package types
+
+import "strings"
+
+const (
+	ModuleName   = "tokenfactory"
+	StoreKey     = ModuleName
+	RouterKey    = ModuleName
+	QuerierRoute = ModuleName
+
+	ActionCreateDenom      = "create_denom"
+	ActionMint             = "tf_mint"
+	ActionBurn             = "tf_burn"
+	ActionSetDenomMetadata = "set_denom_metadata"
+	ActionUpdateParams     = "update_params"
+)
+
+// KeySeparator is used to combine parts of the keys in the store
+const KeySeparator = "|"
+
+var (
+	DenomAuthorityMetadataKey = "authoritymetadata"
+	DenomsPrefixKey           = "denoms"
+	CreatorPrefixKey          = "creator"
+	AdminPrefixKey            = "admin"
+	ParamsPrefixKey           = "params"
+)
```

Comment:
@RiccardoM Do you think we should break the key prefix structure to be as other modules inside Desmos?

Reply:
What do you mean?

Reply:
It is copied from osmosis, should we use `[]byte{0x0}` as prefix instead of using key "denoms"?

Reply:
I think it's fine to use their prefix for forward compatibility here. No need to change them :+1: Also because we won't have many denoms being created (due to the creation fee) so it's not going to impact much the overall storage used.